src/WCSimConstructPMT.cc

```diff
@@ -379,63 +379,25 @@ G4LogicalVolume* WCSimDetectorConstruction::ConstructPMT(G4String PMTName, G4Str
     ////////////////////
     /// 1-PMT support //
     ////////////////////
-    //54mm is position of full PMT support wrt inside of pressure vessel.
-    G4Cons * solidWCPMTsupport =
-      new G4Cons("WCPMTsupport",
-		 0.,                                                                      //rmin1
-		 tan(mPMT_pmt_openingAngle)*(vessel_radius_curv - mPMT_outer_material_d - pmtModuleHeight),      //rmax1
-		 0.,                                                                      //rmin2
-		 tan(mPMT_pmt_openingAngle)*(vessel_radius_curv - mPMT_outer_material_d
-			       -expose - dist_pmt_vessel),                                //rmax2
-		 (pmtModuleHeight - expose - dist_pmt_vessel)/2,                                   //h/2
-		 0.0*deg,                                                                 //phiStart
-		 360.0*deg);                                                              //Deltaphi
-
-    G4LogicalVolume* logicWCPMTsupport =
-      new G4LogicalVolume(solidWCPMTsupport,
-			  G4Material::GetMaterial("Blacksheet"), 
-			  "WCPMTsupport",
-			  0,0,0);
-
-    new G4LogicalSkinSurface("FoamLogSkinSurface",logicWCPMTsupport,OpGelFoamSurface);
-
-    new G4PVPlacement(0,
-		      G4ThreeVector(0, 0, wcpmt_z_offset+(pmtModuleHeight - expose - dist_pmt_vessel)/2),
-		      logicWCPMTsupport,
-		      "WCPMTsupport",
-		      logicWCPMT,
-		      false,
-		      0,
-		      checkOverlaps);
 
     // Visualize
     G4VisAttributes* WCPMTVisAtt_sup = new G4VisAttributes(G4Colour(0.3,0.3,0.3));
     WCPMTVisAtt_sup->SetForceSolid(true);
-    logicWCPMTsupport->SetVisAttributes(WCPMTVisAtt_sup);
```

Comment:
What's this?

Reply:
In the default WCSim geometry, there are two separate support structures, one for PMT and one for the reflector. So to adapt the Solidworks design I used polycone shape from GEANT4 with which I could define the entire support structure as one. But this part of the code for all the geometries except WCSim's default geometry is hardcoded keeping the CAD drawings as reference.